toolbox/src/SetReferences.cpp

```diff
@@ -231,6 +264,26 @@ bool SetReferences::initialize(BlockInformation* blockInfo)
         // This vector caches the target joint positions.
         pImpl->previousReferenceVocabCmPosition.resize(dofs);
     }
+    else if (pImpl->referenceType == impl::ReferenceType::MOTOR_BEMF
+             || pImpl->referenceType == impl::ReferenceType::MOTOR_KTAU) {
+        // Get the interface
+        yarp::dev::ITorqueControl* iTorqueControl = nullptr;
+        if (!getRobotInterface()->getInterface(iTorqueControl) || !iTorqueControl) {
+            wbtError << "Failed to get ITorqueControl interface.";
+            return false;
+        }
+        // Resize the vectors
+        pImpl->motorParams.resize(dofs);
+        pImpl->defaultMotorParams.resize(dofs);
+        // Get the default values
+        for (unsigned m = 0; m < dofs; ++m) {
+            if (!iTorqueControl->getMotorTorqueParams(m, &pImpl->defaultMotorParams[m])) {
```

Comment:
If I am not wrong,  this will cause a blocking RPC call (see https://github.com/robotology/yarp/blob/devel/src/libYARP_dev/src/devices/RemoteControlBoard/RemoteControlBoard.cpp#L2613). I am not sure, but I suspect that the additional delay is not compatible with the real-time behavior of the controller.